packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -37,13 +38,19 @@ class ProductPage extends StatefulWidget {
   State<ProductPage> createState() => _ProductPageState();
 }
 
-class _ProductPageState extends State<ProductPage> {
+class _ProductPageState extends State<ProductPage> with TraceableClientMixin {
   late Product _product;
   late ProductPreferences _productPreferences;
   late ScrollController _scrollController;
   bool _mustScrollToTheEnd = false;
   bool scrollingUp = true;
 
+  @override
+  String get traceName => 'Opened product page for ${widget.product.barcode}';
```

Comment:
```suggestion
  String get traceName => '${widget.product.barcode} is the product barcode';
```

Maybe I'm wrong, but I prefer to see the interesting part first.
Makes me think about the German or Japanese languages, when too often you have to wait for the end of the sentence to know one of the most important part of the sentence: the verb.